### PR TITLE
[pre-commit] Add codespell, and fix some existing typos

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Pylint's own checker checks only python files and not other files (rst/md), it also does not check the content of strings inside the code, thus a bunch of new existing typos fixes were made.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ repos:
     hooks:
       - id: trailing-whitespace
         exclude: tests(/\w*)*/functional/t/trailing_whitespaces.py|tests/pyreverse/data/.*.html|doc/data/messages/t/trailing-whitespace/bad.py
+      #      - id: file-contents-sorter # commented out because it does not preserve comments order
```
Comment (DanielNoord) [inline .pre-commit-config.yaml:10]: Should we just not add it then?

File: custom_dict.txt [hunk 2]
```diff
@@ -126,8 +127,8 @@ formfeed
 fromlineno
 fullname
 func
-functiøn
 functiondef
+functiøn
```
Comment (DanielNoord) [inline custom_dict.txt:131]: What? haha

File: pyproject.toml [hunk 3]
```diff
@@ -193,3 +193,32 @@ ignore = [
...
+#   - bad encoding
+# pylint/pyreverse/diagrams.py and tests/pyreverse/test_diagrams.py:
+#   - An API from pyreverse use 'classe', and would need to be deprecated
+# pylint/checkers/imports.py:
+#   - 'THIRDPARTY' is a value from isort that would need to be handled even
+#   if isort fix the typo in newer versions
+# tests/functional/m/member/member_checks.py:
+#   - typos are voluntary to create credible 'no-member'
+
+skip = """
```
Comment (DanielNoord) [inline pyproject.toml:212]: There is no way to do inline ignores?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml:10]: It's still useful to sort it and then ``git add -p`` the changes.

File: pyproject.toml (hunk 3 above)
Comment (Pierre-Sassoulas) [reply pyproject.toml:212]: Not in codespell (make sense as it's used in documentation mainly)

File: custom_dict.txt (hunk 2 above)
Comment (Pierre-Sassoulas) [reply custom_dict.txt:131]: Yeah should probably remove that instead of sorting it alphabetically 😅 .

Comment (Pierre-Sassoulas) [reply custom_dict.txt:131]: It's used extensively in the unittest for non ascii name

File: .pre-commit-config.yaml (hunk 1 above)
Comment (DanielNoord) [reply .pre-commit-config.yaml:10]: Yeah but I don't think we should be committing code that is commented out to `main` right?

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml:10]: It prevents us from having to recreate the sorting from zero every time we want to sort the file, Is there any downside apart from the file being a little longer ?

Comment (DanielNoord) [reply .pre-commit-config.yaml:10]: Hmm I think it looks a bit sloppy to have code that you don't intend to use, but I guess it is fine?

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**